app/services/bookings/gitis/contact.rb

```diff
@@ -1,68 +1,76 @@
 module Bookings
   module Gitis
     class Contact
-      include ActiveModel::Model
+      include Entity
 
       attr_reader :id
-      attr_accessor :firstname, :lastname, :email, :phone
-      attr_accessor :building, :street, :town_or_city, :county, :postcode
+      entity_attributes :firstname, :lastname, :emailaddress1, :emailaddress2
+      entity_attributes :address1_line1, :address1_line2, :address1_line3
+      entity_attributes :address1_city, :address1_stateorprovince
+      entity_attributes :address1_postalcode, :phone
+
+      alias_attribute :building, :address1_line1
+      alias_attribute :town_or_city, :address1_city
+      alias_attribute :county, :address1_stateorprovince
+      alias_attribute :postcode, :address1_postalcode
 
       validates :email, presence: true, format: /\A.+@.+\..+\z/
 
       def initialize(crm_contact_data = {})
-        @crm_data     = crm_contact_data
-        @id           = @crm_data['contactid']
-        @firstname    = @crm_data['firstname']
-        @lastname     = @crm_data['lastname']
-        @email        = parse_email_from_crm(@crm_data)
-        @phone        = parse_phone_from_crm(@crm_data)
-        @building     = @crm_data['address1_line1']
-        @street       = parse_street_from_crm(@crm_data)
-        @town_or_city = @crm_data['address1_city']
-        @county       = @crm_data['address1_stateorprovince']
-        @postcode     = @crm_data['address1_postalcode']
+        @crm_data                     = crm_contact_data.stringify_keys
+        self.id                       = @crm_data['contactid']
+        self.firstname                = @crm_data['firstname']
+        self.lastname                 = @crm_data['lastname']
+        self.emailaddress1            = @crm_data['emailaddress1']
+        self.emailaddress2            = @crm_data['emailaddress2']
+        self.phone                    = parse_phone_from_crm(@crm_data)
+        self.address1_line1           = @crm_data['address1_line1']
+        self.address1_line2           = @crm_data['address1_line2']
+        self.address1_line3           = @crm_data['address1_line3']
+        self.address1_city            = @crm_data['address1_city']
+        self.address1_stateorprovince = @crm_data['address1_stateorprovince']
+        self.address1_postalcode      = @crm_data['address1_postalcode']
       end
 
       def address
-        [@building, @street, @town_or_city, @county, @postcode].compact.join(", ")
+        [building, street, town_or_city, county, postcode].compact.join(", ")
+      end
+
+      def email
+        emailaddress2.presence || emailaddress1
+      end
+
+      def email=(address)
+        if emailaddress1.blank?
+          self.emailaddress1 = address
```

Comment:
Not really important but this param being called `address` when there's an `address` method a couple of lines up threw me for a second